### PR TITLE
Support sending a `service` tag

### PR DESCRIPTION
**Pull request by ofek**
We want to feature the notion of service more prominently for integrations so that we can correlate metrics with logs and traces

See:

https://github.com/DataDog/datadog-agent/pull/4877
https://github.com/DataDog/datadog-agent/pull/4895

### CODE REVIEW

File: src/main/java/org/datadog/jmxfetch/Instance.java [hunk 1]
```diff
@@ -140,6 +141,14 @@ public Instance(
             this.refreshBeansPeriod = appConfig.getRefreshBeansPeriod();
         }
 
+        this.service = (String) instanceMap.get("service");
+        if (this.service == null && initConfig != null) {
```
Comment (devinsba) [inline src/main/java/org/datadog/jmxfetch/Instance.java]: Since you check for empty string below do you want to do that here too?

Comment (ofek) [reply src/main/java/org/datadog/jmxfetch/Instance.java]: done!

File: src/main/java/org/datadog/jmxfetch/Instance.java [hunk 2]
```diff
@@ -140,6 +141,14 @@ public Instance(
             this.refreshBeansPeriod = appConfig.getRefreshBeansPeriod();
         }
 
+        this.service = (String) instanceMap.get("service");
+        if ((this.service == null || this.service == "") && initConfig != null) {
```
Comment (remeh) [inline src/main/java/org/datadog/jmxfetch/Instance.java]: I think it'll be better to do:
```java
        if ((this.service == null || this.service.isEmpty()) && initConfig != null) {
```
You may already know but you should not use `==` with string in Java, if you don't want to go with `.isEmpty` you should use `.equals`

File: src/main/java/org/datadog/jmxfetch/Instance.java [hunk 3]
```diff
@@ -140,6 +141,14 @@ public Instance(
...
+            this.service = (String) initConfig.get("service");
+        }
+        if (this.service != null && this.service != "") {
```
Comment (remeh) [inline src/main/java/org/datadog/jmxfetch/Instance.java]: `this.service != null && !this.service.isEmpty()`

File: src/main/java/org/datadog/jmxfetch/Instance.java [hunk 4]
```diff
@@ -667,6 +676,10 @@ private void refreshBeansList() throws IOException {
         }
         tags.add("instance:" + this.instanceName);
 
+        if (this.service != null && this.service != "") {
```
Comment (remeh) [inline src/main/java/org/datadog/jmxfetch/Instance.java]: `this.service != null && !this.service.isEmpty()`

**Pull request merged**